.github/workflows/setup-dev-drive.ps1

```diff
@@ -0,0 +1,62 @@
+# Configures a drive for testing in CI.
+
+# When not using a GitHub Actions "larger runner", the `D:` drive is present and
+# has similar or better performance characteristics than a ReFS dev drive.
+# Sometimes using a larger runner is still more performant (e.g., when running
+# the test suite) and we need to create a dev drive. This script automatically
+# configures the appropriate drive.
+
+# Note we use `Get-PSDrive` is not sufficient because the drive letter is assigned.
+if (Test-Path "D:\") {
+    Write-Output "Using existing drive at D:"
+    $Drive = "D:"
+} else {
+	# The size (20 GB) is chosen empirically to be large enough for our
+	# workflows; larger drives can take longer to set up.
+	$Volume = New-VHD -Path C:/ruff_dev_drive.vhdx -SizeBytes 20GB |
```

Comment:
We can probably get away with a smaller drive for Ruff.

Reply:
I actually tried 5 GB then 10 GB and both ran out of space during compilation.